metrics/metrics.go

```diff
@@ -0,0 +1,165 @@
+package metrics
+
+import (
+	"encoding/json"
+	"fmt"
+	log "github.com/Sirupsen/logrus"
+	"github.com/rcrowley/go-metrics"
+	"net/http"
+	"time"
+)
+
+type skipperMetrics map[string]interface{}
+
+// Options for initializing metrics collection.
+type Options struct {
+	// Network address where the current metrics values
+	// can be pulled from. If not set, the collection of
+	// the metrics is disabled.
+	Listener string
+
+	// Common prefix for the keys of the different
+	// collected metrics.
+	Prefix string
+
+	// If set, garbage collector metrics are collected
+	// in addition to the http traffic metrics.
+	EnableDebugGcMetrics bool
+
+	// If set, Go runtime metrics are collected in
+	// addition to the http traffic metrics.
+	EnableRuntimeMetrics bool
+}
+
+const (
+	KeyRouteLookup     = "routelookup"
+	KeyFilterRequest   = "filter.%s.request"
+	KeyFiltersRequest  = "filters.request.%s"
+	KeyProxyBackend    = "backend.%s"
```

Comment:
filters (plural) is the aggregate for all of them where the suffix (%s) will be the route
filter (singular) is the measurement for a specific filter (%s)
Both are measured in the request and response